stan/math/opencl/get_kernel_arg.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef STAN_MATH_OPENCL_GET_KERNEL_ARG_HPP
+#define STAN_MATH_OPENCL_GET_KERNEL_ARG_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <CL/cl.hpp>
+#include <type_traits>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+
+template <typename T>
+inline const T& get_kernel_arg(const T& t) {
+  return t;
+}
+
+inline const cl::Buffer& get_kernel_arg(const matrix_cl& m) {
```

Comment:
`kernel_arg` seems a little ambiguous - kernels also take in ranges and flags, no?

Reply:
Yep! I like your `maybe_get_buffer()` or something like that

Reply:
So I ended up leaving this name. I feel like `get_kernel_arg` conveys 'this returns back the underlying arguments that should go to the kernel. So in cases of an `int`, it's an `int`. For matrix_cl it grabs the buffer.

Reply:
Wugh forgot to add docs for this tho'